host/entry.go

```diff
@@ -0,0 +1,293 @@
+// Copyright 2018 The ZikiChomgo Authors. All rights reserved.  Use of this source
+// code is governed by a license that can be found in the License file.
+
+package host
+
+import (
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	"zikichombo.org/sio/libsio"
+	"zikichombo.org/sound"
+	"zikichombo.org/sound/ops"
+	"zikichombo.org/sound/sample"
+)
+
+// Entry is the type of a connection to host sound I/O capabilities.
+//
+// Entries may either be high level (eg AudioFlinger) or low level (eg ALSA hw).
+//
+// Multiple entries may exist for a given host.  see Names() in the relevant
+// entry_{runtime.GOOS}.go file for details.
+type Entry interface {
+	// Name returns the name of the entry and should be a valid
+	// name for the host.
+	Name() string
+	// DefaultForm returns the default form.  In case one form is not
+	// implemented or uniform accross all supported functionality (input, output, device),
+	// output default should be returned.
+	DefaultForm() sound.Form
+
+	// DefaultSampleCodec returns the default sample codec.  In case one sample codec
+	// is not implemented or uniform accross all supported functionality, output default
+	// should be returned.
+	DefaultSampleCodec() sample.Codec
+
+	// DefaultBufSize returns the default size of the buffer provided to
+	// the caller for interactions.
+	DefaultBufSize() int
+
+	// CanOpenSource returns true if OpenSource does not return
+	// ErrUnspported.
+	CanOpenSource() bool
```

Comment:
Could you elaborate when this is needed? Is it strictly required to have in the main interface? Perhaps an assertion could be used e.g.

```go
if e, ok := entry.(interface { CanOpenSource() bool ); ok {
   e.CanOpenSource()
}
```

Reply:
Oh, I see now. I thought `CanOpenSource` had to do with open source software somehow (e.g. having valid transitive licenses), not *source* in the audio sense. Thus I was confused.